widget/ui/src/containers/SwapInput/SwapInput.tsx

```diff
@@ -1,27 +1,35 @@
-import type { SwapInputProps } from './SwapInput.types';
+import type { BaseProps, SwapInputProps } from './SwapInput.types';
 
-import React from 'react';
+import React, { useRef } from 'react';
 
 import {
   Button,
   Divider,
   PriceImpact,
   Skeleton,
+  Tooltip,
   Typography,
 } from '../../components';
+import { useTextOverflowDetector } from '../../hooks';
 
 import { Container, InputAmount } from './SwapInput.styles';
 import { TokenSection } from './TokenSection';
 
 export function SwapInput(props: SwapInputProps) {
+  const tooltipContentRef = useRef<HTMLSpanElement | null>(null); // Specify the type
+  const isTruncated = useTextOverflowDetector<BaseProps['price']['usdValue']>(
```

Comment:
I'm not okay with using a js solution. I think we can fix it using CSS only.

Reply:
I used it to show tooltips for long text that we want to cut them. I removed the tooltip.